code/modules/mob/mob_helpers.dm

```diff
@@ -584,11 +584,19 @@ var/list/intents = list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM)
 					break
 			if(newname)
 				break	//That's a suitable name!
-			to_chat(src, "Sorry, that [role]-name wasn't appropriate, please try another. It's possibly too long/short, has bad characters or is already taken.")
+			to_chat(src, "Sorry, that [role] name wasn't appropriate, please try another. It's possibly too long/short, has bad characters or is already taken.")
 
 		if(!newname)	//we'll stick with the oldname then
 			return
 
+		if(ishuman(src))
+			var/mob/living/carbon/human/H = src
+			if(H.cannamepick == FALSE)

```

Comment:
`!cannamepick`

Reply:
Wasnt sure if byond had that operator, so i stood on the safe side.